pkg/build/path_rewriter.go

```diff
@@ -33,10 +33,15 @@ func makeAbsWithParent(parent, obj *url.URL) *url.URL {
 	if path.IsAbs(obj.Path) {
 		return obj
 	}
+	parentDir := path.Dir(parent.Path)
```

Comment:
If parent is the path of a component, why would it every be a directory?

Reply:
Maybe I'm not following.

parent.Path is the full path of the component. This just grabs the directory. `foo/bar/biff-component.yaml => foo/bar`

Reply:
I'm wondering where we would ever hit the if below this. i.e. why wasn't this correct before?

Reply:
Man, I can't remember. this pr is almost two months old. XD